File: Old-School-Ted/src/main/java/com/example/OldSchoolTeed/dto/AddItemRequest.java
package com.example.OldSchoolTeed.dto;

import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class AddItemRequest {

    @NotNull(message = "El ID del producto no puede ser nulo")
    private Integer productoId;

    @NotNull(message = "La cantidad no puede ser nula")
    @Min(value = 1, message = "La cantidad debe ser al menos 1")
    private Integer cantidad;


    private PersonalizacionRequest personalizacion;
    private ParcheRequest parche;

    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class PersonalizacionRequest {
        private String tipo;
        private String nombre;
        private String numero;
        private BigDecimal precio;
    }

    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class ParcheRequest {
        private String tipo;
        private BigDecimal precio;
    }
}

File: Old-School-ted-frontend/src/app/models/add-item-request.ts
export interface AddItemRequest {
    productoId: number;
    cantidad: number;
    
   
    personalizacion?: {
        tipo: string;   // 'Leyenda' o 'Custom'
        nombre: string; // Ej: "MESSI"
        numero: string; // Ej: "10"
        precio: number; // Ej: 97.00
    };
    parche?: {
        tipo: string;   // 'UCL' o 'LaLiga'
        precio: number; // Ej: 73.00
    };
}